setup.py

```diff
@@ -22,24 +22,19 @@
 import os
 import re
 import sys
+import subprocess
 
 from setuptools import find_packages, setup
 
 PKG = "hy"
 VERSIONFILE = os.path.join(PKG, "version.py")
-verstr = "unknown"
 try:
-    verstrline = open(VERSIONFILE, "rt").read()
-except EnvironmentError:
-    pass  # Okay, there is no version file.
-else:
-    VSRE = r"^__version__ = ['\"]([^'\"]*)['\"]"
-    mo = re.search(VSRE, verstrline, re.M)
-    if mo:
-        __version__ = mo.group(1)
-    else:
-        msg = "if %s.py exists, it is required to be well-formed" % VERSIONFILE
-        raise RuntimeError(msg)
+    __version__ = subprocess.check_output([
+        "git", "describe", "--tags", "--dirty"]).decode('ASCII').strip()
```

Comment:
@Kodiologist Oh wow, I didn't realize that. I'd prefer supported it personally, since it just adds one extra keyword argument. Though I suppose it's an extremely rare use case...